api4/user.go

```diff
@@ -784,6 +784,19 @@ func getUsers(c *Context, w http.ResponseWriter, r *http.Request) {
 			c.SetPermissionError(model.PermissionReadChannel)
 			return
 		}
+
+		if !*c.App.Config().TeamSettings.ExperimentalViewArchivedChannels {
```

Comment:
The check is inside the `if inChannelId != ""` block, so this should only apply for those cases that include an `inChannelId`

Reply:
The problem is the `if inChannelId != ""` is part of an `else` statement, so if any of the previous ifs is true, this is not going to be checked ever, so you can have a `inChannelId == ""` and at the same time, reach that line, for example, having an `inTeamId != ""` and `inChannelId == ""`.

Reply:
It should have no effect as the only time `inChannelId` is used is inside that conditional, and if it's set but the logic doesn't enter on that specific conditional block, it would be ignored.

In any case, I'm happy to move the check to be more generic so we don't have to change it if the logic ever changes. Updated on d791b95, PTAL :)